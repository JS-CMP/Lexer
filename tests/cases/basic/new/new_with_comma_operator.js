function Foo() {
    this.value = 10
}

console.log(new (0, Foo)()) // NEW((0, Foo))
