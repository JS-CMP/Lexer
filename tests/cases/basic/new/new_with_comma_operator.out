#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {
    JS::Any Foo;
    Foo = JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {

        {
            thisArg[u"value"] = JS::Any(10);
        }
        return JS::Any();
    }, 0, u"Foo"));
    console[u"log"](NEW((JS::Any(0) , Foo)));
    return 0;
}
